src/tensors/factorizations/factorizations.jl

```diff
@@ -0,0 +1,205 @@
+# Tensor factorization
+#----------------------
+# using submodule here to import MatrixAlgebraKit functions without polluting namespace
+module Factorizations
+
+export eig, eig!, eigh, eigh!
+export tsvd, tsvd!, svdvals
+export leftorth, leftorth!, rightorth, rightorth!
+export leftnull, leftnull!, rightnull, rightnull!
+export leftpolar, leftpolar!, rightpolar, rightpolar!
+export copy_oftype, permutedcopy_oftype
+export TruncationScheme, notrunc, truncbelow, truncerr, truncdim, truncspace
+
+using ..TensorKit
+using ..TensorKit: AdjointTensorMap, SectorDict, OFA, blocktype, foreachblock
+using ..MatrixAlgebra: MatrixAlgebra
+
+using LinearAlgebra: LinearAlgebra, BlasFloat
+import LinearAlgebra: eigen, eigen!, isposdef, isposdef!, ishermitian
+
+using TensorOperations: Index2Tuple
+
+using MatrixAlgebraKit
+using MatrixAlgebraKit: AbstractAlgorithm, TruncatedAlgorithm, TruncationStrategy,
+                        NoTruncation, TruncationKeepAbove, TruncationKeepBelow,
+                        TruncationIntersection, TruncationKeepFiltered
+import MatrixAlgebraKit: select_algorithm,
+                         default_qr_algorithm, default_lq_algorithm,
+                         default_eig_algorithm, default_eigh_algorithm,
+                         default_svd_algorithm, default_polar_algorithm,
+                         copy_input, check_input, initialize_output,
+                         qr_compact!, qr_full!, qr_null!, lq_compact!, lq_full!, lq_null!,
+                         svd_compact!, svd_full!, svd_trunc!,
+                         eig_full!, eig_trunc!, eigh_full!, eigh_trunc!,
+                         left_polar!, left_orth_polar!, right_polar!, right_orth_polar!,
+                         left_null_svd!, right_null_svd!,
+                         left_orth!, right_orth!, left_null!, right_null!,
+                         truncate!, findtruncated, findtruncated_sorted,
+                         diagview
+
+include("utility.jl")
+include("interface.jl")
+include("implementations.jl")
+include("matrixalgebrakit.jl")
+include("truncation.jl")
+include("deprecations.jl")
+
+"""
+    isposdef(t::AbstractTensor, (leftind, rightind)::Index2Tuple) -> ::Bool
+
+Test whether a tensor `t` is positive definite as linear map from `rightind` to `leftind`.
+
+If `leftind` and `rightind` are not specified, the current partition of left and right
+indices of `t` is used. In that case, less memory is allocated if one allows the data in
+`t` to be destroyed/overwritten, by using `isposdef!(t)`. Note that the permuted tensor on
+which `isposdef!` is called should have equal domain and codomain, as otherwise it is
+meaningless.
+"""
+function LinearAlgebra.isposdef(t::AbstractTensorMap, (p₁, p₂)::Index2Tuple)
+    tcopy = permutedcopy_oftype(t, factorisation_scalartype(isposdef, t), p)
+    return isposdef!(tcopy)
+end
+function LinearAlgebra.isposdef(t::AbstractTensorMap)
+    tcopy = copy_oftype(t, float(scalartype(t)))
```

Comment:
Should the second argument also be `factorisation_scalartype(isposdef, t)`?

Reply:
I went ahead and implemented this as such, if you disagree I'm happy to change it back.